addon/components/freestyle/bootstrap-button.js

```diff
@@ -0,0 +1,32 @@
+import Ember from 'ember';
+import layout from '../../templates/components/freestyle/bootstrap-button';
+import { task, timeout, didCancel, restartable } from 'ember-concurrency';
+
+export default Ember.Component.extend({
+  layout,
+  fakeSimpleClickTask: task(function * (name) {
+    alert(`You clicked the ${name} button`);
+  }).restartable(),
+
+  fakeSlowClickTask: task(function * (name) {
+    yield timeout(1000);
+    alert(`You clicked the slow ${name} button`);
+  }).restartable(),
+
+  actions: {
+    simpleClick(name) {
+      return this.get('fakeSimpleClickTask').perform(name).catch(e => {
+        if (!didCancel(e)) {
+          throw e;
+        }
+      });
+    },
+    slowClick(name){
```

Comment:
This is just a question for my personal knowledge lack...what is the difference between a simple and slow click?

Reply:
I think this is just to demonstrate the async behavior for the button, by having one button on the Freestyle page that takes 1 second to complete its task.